colossalai/auto_parallel/tensor_shard/node_handler/strategy/strategy_generator.py

```diff
@@ -151,11 +153,30 @@ def _compute_size_in_bytes(self, strategy: ShardingStrategy, key: str):
         size_per_elem_bytes = torch.tensor([], dtype=dtype).element_size()
         return reduce(operator.mul, sharded_shape) * size_per_elem_bytes
 
-    @abstractmethod
     def generate(self) -> List[ShardingStrategy]:
         """
         Generate all possible sharding strategies for this operation.
         """
+        strategies = self.collate_strategies()
+
+        # some strategies may be None as ignore_sharding_exception may return None
+        # when ShardingSpecException occurs.
+        # thus, remove those None values
+        strategies = [strategy for strategy in strategies if strategy]
```

Comment:
Need to check whether strategies is a list or not.

Reply:
As indicated by the type hinting of `collate_strateties`, it expects it to be a list of strategies.

Reply:
I see